docs/claimantmodel/Maps.md

```diff
@@ -0,0 +1,115 @@
+
+The spec for canonicalization should be precise and unambiguous so that it can be implemented in precisely the same way on all clients. Any discrepancy in implementation is a vector for presenting different Claims to different actors. This type of issue is insidious because the data structure verification would pass, giving a false sense of confidence. An attacker could effectively use this to provide a split-view without all the pains required to maintain root commitments that could not be reconciled.
+
+Non-canonical forms are not uniquely a problem for maps. Having ambiguous forms could be a problem for Claim Verifiers in the Log Claimant model too, but there Verifiers have the raw Subject and can canonicalize it, or query for substrings/prefixes, etc.
+
+### Map Functions (deriving Claim Subjects from a Claim)
+
+Using a CSM is simplest in the case where there is a single Subject for every Claim. However, in the general case there are multiple Claim Subjects entailed by a Claim. This apparently minor difference has a large impact on the amount of work required in order to maintain discoverability.
+
+```
+mapFn(Claim) []ClaimSubject
+```
+
+We’ll use the signature above when describing how Subjects are derived from Claims. For a Believer to trust that a given Claim is discoverable, they must check that the Claim is committed to under all Subjects returned by mapFn.
+
+The reasoning behind this is best understood through example. If we look at the Subjects entailed by a Certificate for `*.foo.example.co.uk` we might derive:
+ * `foo.example.co.uk.`
+ * `example.co.uk.`
+ * `co.uk.`
+ * `uk.`
+ * `.`
+
+This set of Subjects must have a non-empty intersection with the set of subjects that the Domain Owner is expecting. If this Domain Owner does not have any subdomains and expects to be checking only `example.co.uk.` then a believer convinced only of the inclusion of `foo.example.co.uk.` and no others will have trusted a Claim which cannot be discovered.
+
+This example also highlights that mapFn should not return too many Subjects. While the empty domain `.` may be derived from all Claims, doing so entails a map key which is effectively committing to all entries that would have been in the log. If this is the only mechanism that permits discoverability then a log should have been chosen instead of a map.
+
+### CSM in the Claimant Model
+#### Log-Backed Maps & Mogs
+
+If the map is constructed from a Claim Log then we have the following:
+
+<dl>
+<dt>Claim<sup>CSM</sup></dt>
+<dd><i>"Applying $mapFn to the logged data in $LogCheckpoint results in the map with $rootHash"</i></dd>
+<dt>Statement<sup>CSM</sup></dt>
+<dd>Map Checkpoint (sometimes SMR, or Signed Map Root)</dd>
```

Comment:
which itself contains/commits to at least $LogCheckpoint.Size

Reply:
I'm going to merge this as currently written. Happy to continue this discussion and make changes in future PRs.